Dart/src/com/jetbrains/lang/dart/ide/runner/server/webdev/DartWebdevConfiguration.java

```diff
@@ -56,6 +59,28 @@ public RunProfileState getState(@NotNull final Executor executor,
     return new DartWebdevRunningState(env);
   }
 
+  @Nullable
+  @Override
+  public String suggestedName() {
+    // Attempt to compute the relative path to the html file, i.e. some "web/index.html"
+    // If not successful, return at least the file name, i.e. some "index.html"
+    final String htmlFilePath = myParameters.getHtmlFilePath();
+    String htmlFilePathRelativeFromWorkingDir = null;
+    try {
+      VirtualFile workingDir = myParameters.getWorkingDirectory(getProject());
+      htmlFilePathRelativeFromWorkingDir =
+        htmlFilePath.startsWith(workingDir.getPath() + "/") ? htmlFilePath.substring(workingDir.getPath().length() + 1) : null;
+    }
+    catch (RuntimeConfigurationError ignore) {
+    }
+
+    if (StringUtil.isNotEmpty(htmlFilePathRelativeFromWorkingDir)) {
+      return htmlFilePathRelativeFromWorkingDir;
+    }
+
+    return StringUtil.isEmpty(htmlFilePath) ? null : PathUtil.getFileName(htmlFilePath);
```

Comment:
Returning not-null here enables some magic: the name of the manually created run configuration changes as user types in or selects an HTML file. But once the name is manually edited, it's not auto-updated anymore.
I've changed null to `"Dart web app"`.